libnmstate/nm/sriov.py

```diff
@@ -43,11 +90,67 @@ def create_setting(iface_state, base_con_profile):
         if not sriov_setting:
             sriov_setting = nmclient.NM.SettingSriov.new()
 
+        vfs_config = sriov_config.get(Ethernet.SRIOV.VFS_SUBTREE, [])
+        vf_object_ids = {vf.get_index() for vf in sriov_setting.props.vfs}
+        vf_config_ids = {
+            vf_config[Ethernet.SRIOV.VFS.ID] for vf_config in vfs_config
```

Comment:
Seems that there is a hidden assumption that the user/client includes this ID in the desired state.
I think we need to make sure it is included, and if it is not, to raise an error. Seems to fit the validator module.

Reply:
It is enforced in the schema.